modules/sqlite_helpers.py

```diff
@@ -48,45 +49,55 @@ def insert_url(sqlite_file: str, url: str, alias: str):
         logger.exception("Inserting url had an error")
         return False
 
-def get_urls(sqlite_file: str): #returns all urls in the table
+def get_urls(sqlite_file, page=0, limit=ROWS_PER_PAGE, search=None):
     db = sqlite3.connect(sqlite_file)
     cursor = db.cursor()
     
-    sql = "SELECT * from urls order by created_at desc"
-    cursor.execute(sql)
+    offset = page * limit
+    if search:
+        sql = """
+        SELECT * FROM urls 
+        WHERE LOWER(alias) LIKE LOWER(?) 
+        OR LOWER(url) LIKE LOWER(?)
+        LIMIT ? OFFSET ?
+        """
+        cursor.execute(sql, ('%' + search + '%', '%' + search + '%', limit, offset))
+    else:
+        sql = "SELECT * FROM urls LIMIT ? OFFSET ?"
+        cursor.execute(sql, (limit, offset))
```

Comment:
can we do something like this instead?

```py
maybe_where = ""
if search:
    maybe_where = f"""
        WHERE LOWER(alias) LIKE LOWER(%{search}%) 
        OR LOWER(url) LIKE LOWER(%{search}%)
    """
sql = f"""
    SELECT * FROM urls 
    {maybe_where}
    LIMIT {limit} OFFSET {offset}
"""
cursor.execute(sql, ('%' + search + '%', '%' + search + '%', limit, offset))
result = cursor.fetchall()
# ...
```

let me know if we can use fstrings with sqlite

Reply:
Done